src/pages/MergePositions/Contents.tsx

```diff
@@ -7,122 +7,152 @@ import { Button } from 'components/buttons/Button'
-        .then((token) => {
-          if (!cancelled) {
-            setCollateralToken(token)
-          }
-        })
-        .catch((err) => {
-          logger.error(err)
-        })
-    }
-    return () => {
-      cancelled = true
-    }
-  }, [positions, networkConfig, provider, mergeablePositions])
-
-  const maxBalance = useMemo(
-    () => (mergeablePositions && balances.length ? minBigNumber(balances) : ZERO_BN),
-    [balances, mergeablePositions]
-  )
+  const [mergeResult, setMergeResult] = useState<string>('')
 
   const [amount, setAmount] = useState<BigNumber>(ZERO_BN)
-  const amountChangeHandler = useCallback((value: BigNumber) => {
-    setAmount(value)
-  }, [])
+  const [collateralToken, setCollateralToken] = useState<Maybe<Token>>(null)
+  const [clearFilters, setClearFilters] = useState(false)
 
-  const useWalletHandler = useCallback(() => {
-    if (mergeablePositions && maxBalance.gt(ZERO_BN)) {
-      setAmount(maxBalance)
+  const isConditionResolved = useIsConditionResolved(conditionId)
+
+  useEffect(() => {
+    const getCollateral = async () => {
+      try {
+        if (position) {
+          const token = await getTokenSummary(networkConfig, provider, position.collateralToken)
+          setCollateralToken(token)
+        }
+      } catch (err) {
+        logger.error(err)
+      }
+      return null
     }
-  }, [maxBalance, mergeablePositions])
+
+    getCollateral()
+  }, [networkConfig, provider, position])
 
   const decimals = useMemo(() => (collateralToken ? collateralToken.decimals : 0), [
     collateralToken,
   ])
 
-  const disabled = useMemo(
-    () =>
-      status === Status.Loading ||
-      positionsErrors.length > 0 ||
-      conditionErrors.length > 0 ||
-      balancesErrors.length > 0 ||
-      !canMergePositions ||
-      amount.isZero(),
-    [canMergePositions, amount, status, positionsErrors, conditionErrors, balancesErrors]
+  const clearComponent = useCallback(() => {
+    setConditionId(null)
+    setPosition(null)
+    setConditionIds([])
+    setIsLoadingConditions(false)
+    setMergeablePositions([])
+    setIsLoadingMergeablePositions(false)
+    setSelectedPositions([])
+    setSelectedBalancePositions([])
+    setMergeResult('')
+    setAmount(ZERO_BN)
+    setMergeResult('')
+    setTransactionStatus(Remote.notAsked<Maybe<boolean>>())
+  }, [])
+
+  const onConditionSelect = useCallback((conditionId: string) => {
+    setConditionId(conditionId)
+    logger.log(conditionId)
+  }, [])
+
+  const onMergeableItemClick = useCallback(
+    (item: MergeablePosition, index: number) => {
+      logger.log(item, index)
+      const { position } = item
+      const existPosition = selectedPositions.find(
+        (selectedPosition) => selectedPosition.id === position.id
+      )
+      if (!existPosition) {
+        setSelectedPositions([...selectedPositions, position])
+        setConditionId(position.conditions[0].conditionId)
+      }
+    },
```

Comment:
I'm putting this comment here but I'm not sure where is the error, the merge preview is not using the selected position, checking at `MergePreview` with `console.log(positions.map((p) => p.id))` only the ones checked from "merge with..." box. It's visible on the preview because there must be n-1 `|` where n is the number of positions selected, and we have n-2 'pipes'.